lib/editor/components/timetable/EditableCell.js

```diff
@@ -55,120 +55,127 @@ const renderCell = (
 /**
  * A component to handle the editing of a cell in a timetable editor
  */
-export default class EditableCell extends Component<Props, State> {
-  componentWillMount () {
-    this.setState({
-      isEditing: this.props.isEditing,
-      isFocused: false,
-      edited: false,
-      data: this.props.data,
-      originalData: this.props.data
-    })
-  }
-
-  cellInput = null
+function EditableCell (props: Props) {
+  const [state, setState] = useState<State>({
+    isEditing: props.isEditing,
+    isFocused: false,
+    edited: false,
+    data: props.data,
+    originalData: props.data
+  })
 
   /**
    * The component may receive data from a save event or
    * editing can be changed by a change in the activeCell in the TimetableGrid
    */
-  componentWillReceiveProps (nextProps: Props) {
-    if (this.props.data !== nextProps.data) {
-      this.setState({data: nextProps.data})
+  useEffect(() => {
+    setState((prevState) => ({
+      ...prevState,
+      data: props.data
+    }))
+    if (state.isEditing !== props.isEditing) {
+      setState((prevState) => ({
+        ...prevState,
+        isEditing: props.isEditing
+      }))
     }
-    if (this.state.isEditing !== nextProps.isEditing) {
-      this.setState({isEditing: nextProps.isEditing})
-    }
-  }
+  }, [props.data, props.isEditing])
 
-  cancel = () => {
-    this.setState({
+  const cancel = () => {
+    setState((prevState) => ({
+      ...prevState,
       isEditing: false,
       isFocused: false,
-      data: this.props.data
-    })
-    this.props.onStopEditing()
+      data: props.data
+    }))
+    props.onStopEditing()
   }
 
-  beginEditing () {
-    const {columnIndex, rowIndex, setActiveCell} = this.props
+  const beginEditing = () => {
+    const {columnIndex, rowIndex, setActiveCell} = props
```

Comment:
Why are these variables defined?

Reply:
The console was mad when I didn't define the variables since I switched everything to a functonal component